app/assets/javascripts/courses.js

```diff
@@ -1,98 +1,7 @@
 var ready;
 
 ready = function() {
-
-	function sortProfessors(selector, attrName) {
-
-		return $($(selector).toArray().sort(function(professorA, professorB) {
-
-			// Compare two professors by a given stat
-			retVal = compareProfessors(professorA, professorB, attrName);
-
-			// Handle tie
-			if (retVal == 0) {
-				var otherSortOptions = [".course-rating", ".course-difficulty", ".course-gpa"];
-
-				for (var i = 0; i < otherSortOptions.length; i++) {
-					// Get another stat
-					if (attrName != otherSortOptions[i]) {
-
-						// Compare by that stat
-						retVal = compareProfessors(professorA, professorB, otherSortOptions[i]);
-
-						// stop if found a tie breaker
-						if (retVal != 0) {
-							break;
-						}
-						// if three-way tie, return 0
-						if (i == 2 && retVal == 0) {
-							return 0;
-						}
-					}
-				}
-			}
-
-			return retVal;
-
-		}));
-	}
-
-	// Compares two professor panel elements by a given stat (id)
-	function compareProfessors(professorA, professorB, stat) {
-		// Get stat to sort by (rating, difficulty, gpa)
-		var aVal = parseFloat($(professorA).find(stat).text()),
-			bVal = parseFloat($(professorB).find(stat).text()),
-			retVal = bVal - aVal;
-
-		// Handle no stat
-		if (isNaN(aVal) && !isNaN(bVal)) {
-			return 1;
-		} else if (isNaN(bVal) && !isNaN(aVal)) {
-			return -1;
-		} else if (isNaN(bVal) && isNaN(aVal)) {
-			retVal = 0;
-		}
-
-		// Sort difficulty ascending
-		if (stat == ".course-difficulty") {
-			retVal = retVal * -1;
-		}
-
-		return retVal;
-	}
-
-
-	// Sort professors by stats
-	$('#prof-sort').change(function() {
-
-		// id of stat to sort by		
-		var sortString = ".course-" + $(this).find('.active')[0].id,
-			// class of which panels are displayed (current semester or all)
-			selectorString = $("#all").parent().hasClass("active") ? ".prof-panel.all" : ".prof-panel.current",
-			// how many panels there are (to know when to trigger the next animation)
-			numPanels = $(selectorString).length,
-			slidPanels = 0,
-			// sorted list of professors
-			profList = sortProfessors($(".prof-panel"), sortString);
-		// if the number of panels displayed is small enough to have smooth animations, animate the change
-		if (numPanels < 50) {
```

Comment:
Should we maybe look into a more performant/less janky transition (perhaps only using the opacity and translation attributes) so we can have the same behavior across all panel numbers? Is there a reason we have this threshold?